### PR TITLE
Feat/398 use poetry

### LINKED ISSUE #398: [Feature Request] Use Poetry for packaging
**Current Behavior**

Pip and conda are very tedious to maintain, and should be avoided at all costs. Maintenance of package dependencies is an NP complete problem. 

**Desired Behavior**

Poetry uses best practices to balance updating packages and keeping them stable for a specific project. 

```
python3 -m venv <directory>
pip install poetry
```

A combination of virtual environment and poetry is something I would highly recommend using. Especially in the main readme of the project as well. 

---

**Screenshots / Mockups**

**Alternatives**

---

### PR DESCRIPTION
**Pull request by IdoKendo**
**Description**

This PR fixes #398 

This PR adjusts `pyproject.toml` to manage dependencies and the like using poetry. It also allows running the development server script using `poetry run devserver`.

The README.md file has also been updated to reflect this change.

### CODE REVIEW

**Comment by sansyrox** [comment]
Thanks for the PR @IdoKendo 😄 

I will take some time to review this. As I want to see how it performs with the CI, my dev setup, and the maturin toolkit.

**Comment by sansyrox** [comment]
@IdoKendo , looks like the CI is failing with this change. Can you have a look, please?

**Comment by IdoKendo** [comment]
> @IdoKendo , looks like the CI is failing with this change. Can you have a look, please?

I've updated the CI to use Poetry as well. it should work now 😃

File: pyproject.toml [hunk 1]
```diff
@@ -35,21 +21,52 @@ classifiers = [
...
+maturin = "0.14.12"
+watchdog = "2.2.1"
+multiprocess = "0.70.14"
+nestd = "0.3.1"
+uvloop = { version = "0.17.0", markers = "sys_platform == 'darwin' or platform_machine == 'aarch_64' or platform_machine == 'x86_84' or platform_machine == 'i686'" }
+jinja2 = { version = "3.0.1", optional = true }
+
+[tool.poetry.extras]
+templating = ["jinja2"]
+
```
Comment (sansyrox) [inline pyproject.toml:39]: love this

**Comment by sansyrox** [comment]
@IdoKendo , this pr looks great! 😄 

There are some merge conflicts. We will need to fix them before merging 😅

**Comment by IdoKendo** [comment]
> @IdoKendo , this pr looks great! 😄
> 
> There are some merge conflicts. We will need to fix them before merging 😅

Sure thing, will fix ✨

**Comment by IdoKendo** [comment]
@sansyrox Fixed the conflicts, but it seems that there are still non-trivial issues with CI actions. Looking into it.

**Comment by sansyrox** [comment]
@IdoKendo , have you tried rebasing the main branch?

**Comment by sansyrox** [comment]
As, I think something else is wrong here. The ci is failing for all the latest prs :(

**Comment by IdoKendo** [comment]
Hey @sansyrox 👋 
Finally had the time to get back to this one, it seems that poetry was a little finicky about paths and relative paths, but seems to all be working now! Let me know if there's any gaps remaining with this change!

**Comment by sansyrox** [comment]
Hey @IdoKendo 👋 

Thank you for your PR 😄 I have had an initial look today. I will have a deep look tomorrow. 😄

File: pyproject.toml [hunk 2]
```diff
@@ -7,13 +7,8 @@ name = "robyn"
...
 description = "A web server that is fast!"
 authors = [{ name = "Sanskar Jethi", email = "sansyrox@gmail.com" }]
-dependencies = [
-  'watchdog == 2.2.1',
-  'multiprocess == 0.70.14',
-  'nestd == 0.3.1',
-# conditional
-  "uvloop>=0.17.0; sys_platform != 'win32' and (sys_platform != 'cygwin' and platform_python_implementation != 'PyPy')"
-  ]
+repository = "https://github.com/sansyrox/robyn"
```
Comment (sansyrox) [inline pyproject.toml:10]: let us rename it to `sparckles/robyn`

File: pyproject.toml [hunk 3]
```diff
@@ -30,21 +25,71 @@ classifiers = [
   "Programming Language :: Python :: Implementation :: CPython",
 ]
 
+[tool.poetry]
+name = "robyn"
+version = "0.36.2"
+description = "A web server that is fast!"
```
Comment (sansyrox) [inline pyproject.toml:31]: We should update it to 

```
About
A High-Performance, Community-Driven, and Innovator Friendly Web Framework with a Rust runtime.
```

File: pyproject.toml [hunk 4]
```diff
@@ -30,21 +25,71 @@ classifiers = [
...
+authors = ["Sanskar Jethi <sansyrox@gmail.com>"]
+repository = "https://github.com/sansyrox/robyn"
```
Comment (sansyrox) [inline pyproject.toml:33]: same `sparckles/robyn`

File: pyproject.toml [hunk 5]
```diff
@@ -30,21 +25,71 @@ classifiers = [
...
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: Implementation :: CPython",
+]
 
-[project.urls]
-"Source Code" = "https://github.com/sansyrox/robyn"
-Issues = "https://github.com/sansyrox/robyn/issues"
-Documentation = "https://robyn.tech/"
+[tool.poetry.urls]
+"Bug Tracker" = "https://github.com/sansyrox/robyn/issues"
```
Comment (sansyrox) [inline pyproject.toml:52]: same `sparckles/robyn`

File: pyproject.toml [hunk 6]
```diff
@@ -30,21 +25,71 @@ classifiers = [
...
 Changelog = "https://github.com/sansyrox/robyn/blob/main/CHANGELOG.md"
```
Comment (sansyrox) [inline pyproject.toml:53]: same `sparckles/robyn`

File: pyproject.toml [hunk 7]
```diff
@@ -30,21 +25,71 @@ classifiers = [
...
+optional = true
+
+[tool.poetry.group.test.dependencies]
+requests = "2.28.2"
+pytest = "7.2.1"
+websocket-client = "1.5.0"
+pytest-codspeed = "1.2.2"
+
+[tool.poetry.scripts]
+devserver = { callable = "integration_tests.base_routes:main" }
```
Comment (sansyrox) [inline pyproject.toml:87]: one final change
can we name it `test_server` or something similar?

We have a dev server for live reloading

**Comment by sansyrox** [comment]
Hey @IdoKendo 👋 

Thank you for the PR 😄 I have just some final suggestions

**Comment by IdoKendo** [comment]
Hey @sansyrox, I changed all the values as suggested, let me know if there's anything more 😄

**Comment by sansyrox** [comment]
Hey @IdoKendo 👋 

Apologies for the late merge. I was fixing some issues with the release CI. Can you please rebase the changes and I will merge them 😄

**Comment by IdoKendo** [comment]
> Hey @IdoKendo 👋
> 
> Apologies for the late merge. I was fixing some issues with the release CI. Can you please rebase the changes and I will merge them 😄

Hey @sansyrox , done 😃

**Review by sansyrox — APPROVED** [review]
LGTM! 😄 

Great work @IdoKendo 🔥

**Pull request merged**